build-tools/src/main/java/software/amazon/awssdk/buildtools/checkstyle/PluralEnumNames.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.buildtools.checkstyle;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A rule that disallows plurality in enum names (eg. it should be "Property.MY_PROPERTY", not "Properties.MY_PROPERTY"). This
+ * also applies the validation to "pseudo-enums", classes with public static fields.
+ */
+public class PluralEnumNames extends AbstractCheck {
+    private static final List<String> DISALLOWED_ENDINGS = Arrays.asList("S", "s");
+    private static final List<String> ALLOWED_ENDINGS = Arrays.asList("Status");
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] { TokenTypes.CLASS_DEF, TokenTypes.ENUM_DEF };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        String classOrEnumName = ast.findFirstToken(TokenTypes.IDENT).getText();
+
+        if (isPlural(classOrEnumName) && isPluralDisallowed(ast)) {
+            log(ast, "Enum or class name should be singular, not plural (Property.VALUE not Properties.VALUE).");
+        }
+    }
+
+    private boolean isPlural(String className) {
+        return DISALLOWED_ENDINGS.stream().anyMatch(className::endsWith) &&
```

Comment:
It's kind of confusing to name them allowed/disallowed since it depends more on the context in which they are used whether they are allowed or not. Perhaps just PLURAL_ENDINGS and NON_PLURAL_ENDINGS or something along those lines?

Reply:
Fixed.